pkg/controllers/workgenerator/override_test.go

```diff
@@ -2496,3 +2759,88 @@ func TestApplyJSONPatchOverride(t *testing.T) {
 		})
 	}
 }
+
+func Test_replaceClusterLabelKeyVariables(t *testing.T) {
+	tests := map[string]struct {
+		cluster   *clusterv1beta1.MemberCluster
+		input     string
+		expected  string
+		expectErr bool
+	}{
+		"No clusterLabelKey variables": {
+			cluster: &clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"region": "us-west-1",
+					},
+				},
+			},
+			input:    "The cluster is in us-west-1",
+			expected: "The cluster is in us-west-1",
+		},
+		"ClusterLabelKey Variable replaced": {
+			cluster: &clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"region": "us-west-1",
+					},
+				},
+			},
+			input:    "The cluster is in ${MEMBER-CLUSTER-LABEL-KEY-region}",
```

Comment:
what if the input is "The cluster is in ${MEMBER-CLUSTER-LABEL-KEY-}"

Reply:
then the key is empty which means it will throw an error, case added.